File: VyacheslavBuchin/term2/block4/ElementaryBash/src/main/kotlin/command/ExitCommand.kt
package command

import channel.Channel
import channel.StringChannel
import exception.ElementaryBashException
import kotlin.system.exitProcess

class ExitCommand(
	override var input: Channel<String> = StringChannel.nullChannel()
) : Command {
	override var output = StringChannel.nullChannel()
	override var error: Channel<String> = StringChannel()

	override fun execute(args: Array<String>): Int {
		if (args.isEmpty())
			exitProcess(0)
		if (args.size > 1)
			throw ElementaryBashException(ElementaryBashException.INVALID_ARGUMENTS, "too many arguments")
		if (args[0].toIntOrNull() == null)
			throw ElementaryBashException(ElementaryBashException.INVALID_ARGUMENTS, "${args[0]}: numeric argument required")
		exitProcess(args[0].toInt())
	}

}

File: Daniil-Kadochnikov/Second Term/Fourth Block/Bash/Commands/ExitCommand.cs
namespace Commands
{
	public class ExitCommand : ICommand
	{
		public string Name { get; } = "exit";

		public string[]? Execute(string[] arguments)
		{
			if (arguments.Length == 0)
				Environment.Exit(0);
			else if (arguments.Length == 1 && int.TryParse(arguments[0], out int argument))
				Environment.Exit(argument);
			else
				Environment.Exit(-1);
			return null;
		}
	}
}